eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/batch/io/MergedAccessionsVariantReader.java

```diff
@@ -0,0 +1,493 @@
+            orgAccMergeIntoMap.forEach((orgAcc, mergeIntoSet) -> {
+                List<EventDocument<IClusteredVariant, Long, ? extends ClusteredVariantInactiveEntity>> listOfEventDocs = cveOps.entrySet().stream()
+                        .filter(entry -> mergeIntoSet.contains(entry.getKey()))
+                        .flatMap(entry -> entry.getValue().values().stream().flatMap(Collection::stream))
+                        .collect(Collectors.toList());
+                Set<Long> newMergedIntoAcc = listOfEventDocs.stream()
+                        .filter(eventDoc -> eventDoc.getEventType().equals(EventType.MERGED))
+                        .map(eventDoc -> eventDoc.getMergedInto())
+                        .collect(Collectors.toSet());
+                Set<Long> deprecatedAccessions = listOfEventDocs.stream()
+                        .filter(eventDoc -> eventDoc.getEventType().equals(EventType.DEPRECATED))
+                        .map(eventDoc -> eventDoc.getAccession())
+                        .collect(Collectors.toSet());
+
+                // filter out already processed accessions before adding them for processing
+                newMergedIntoAcc.removeAll(alreadyProcessedAccessionsMap.get(orgAcc));
+                if (!newMergedIntoAcc.isEmpty()) {
+                    temp2OrgAccMergeIntoMap.put(orgAcc, newMergedIntoAcc);
+                }
+                if (!deprecatedAccessions.isEmpty()) {
+                    deprecatedAccessionsMap.get(orgAcc).addAll(deprecatedAccessions);
+                }
+            });
+
+            // check and report accessions if an accession is both active and also has merged/deprecated operations
+            Set<Long> bothActiveMergedDeprecatedAcc = cveOps.keySet().stream()
+                    .filter(acc -> activeAccSet.contains(acc))
+                    .collect(Collectors.toSet());
+            if (!bothActiveMergedDeprecatedAcc.isEmpty()) {
+                logger.warn("Accession is both Active and has Merged/Deprecated operations: The following accessions are active as well as have merged/deprecated operations: {}", bothActiveMergedDeprecatedAcc);
+            }
+
+            // check and log accessions which are neither active nor has any further merge/deprecated operations
+            orgAccMergeIntoMap.forEach((orgAcc, mergeIntoSet) -> {
+                mergeIntoSet.forEach(acc -> {
+                    if (!activeAccSet.contains(acc) && !cveOps.containsKey(acc)) {
+                        logger.warn("Accession {} is Neither Active nor has any further Merged/Deprecated Operations. Original Accessions {}", acc, orgAcc);
+                    }
+                });
+            });
+
+            // update processing set
+            orgAccMergeIntoMap = temp2OrgAccMergeIntoMap;
+            processingAccSet = orgAccMergeIntoMap.values().stream()
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toSet());
+        }
```

Comment:
Just for readability, I would highlight somehow that this is the end of the processing loop, like a comment saying "Done processing, now generating results"

Reply:
done

Reply:
it's less important now that you've extracted the loop actually... but thanks anyway!